File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = defaultdict(int)
        for x in nums:
            counts[x] += 1

        arr = []
        for num, count in counts.items():
            arr.append([count, num])
        arr.sort()

        output = []
        n = len(arr)
        for i in range(n-1, n-1-k, -1):
            output.append(arr[i][1])
        return output

File: Data Structures & Algorithms/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/submission-1.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        threshold *= k;
        int sum = 0, count = 0;
        for (int r = 0; r < arr.size(); r++) {
            sum += arr[r];
            if (r >= k - 1) {
                if (sum >= threshold) count++;

                sum -= arr[r - k + 1];
            }
        }
        return count;
    }
};